lib/Runtime/Library/JavascriptRegularExpression.cpp

```diff
@@ -635,21 +798,63 @@ namespace Js
                 JavascriptError::ThrowTypeError(scriptContext, JSERR_RegExpExecInvalidReturnType, varName);
             }
 
-            JavascriptOperators::SetProperty(
-                regEx,
-                thisObj,
-                PropertyIds::lastIndex,
-                previousLastIndex,
-                scriptContext,
-                lastIndexOperationFlags);
+            return result;
+        }
+
+        if (regExObj == nullptr)
+        {
+            regExObj = ToRegExp(thisObj, varName, scriptContext);
+        }
+        return RegexHelper::RegexExec(scriptContext, regExObj, string, false);
+    }
+
+    void JavascriptRegExp::RecompilePatternForExecIfNeeded(ScriptContext* scriptContext)
+    {
+        if (!scriptContext->GetConfig()->IsES6RegExSymbolsEnabled())
+        {
+            return;
+        }
+
+        bool isObservable =
+            JavascriptRegExp::HasObservableGlobalFlag(this, scriptContext)
+            || JavascriptRegExp::HasObservableStickyFlag(this, scriptContext);
+        if (!isObservable)
+        {
+            return;
+        }
 
-            return (JavascriptOperators::GetTypeId(result) == TypeIds_Null)
-                ? TaggedInt::ToVarUnchecked(-1)
-                : JavascriptOperators::GetProperty(RecyclableObject::FromVar(result), PropertyIds::index, scriptContext);
+        UnifiedRegex::RegexPattern* pattern = this->GetPattern();
+
+        UnifiedRegex::RegexFlags newFlags = pattern->GetFlags();
+        newFlags = SetRegexFlag(PropertyIds::global, newFlags, UnifiedRegex::GlobalRegexFlag, scriptContext);
+        newFlags = SetRegexFlag(PropertyIds::sticky, newFlags, UnifiedRegex::StickyRegexFlag, scriptContext);
```

Comment:
It is possible. According to [`RegExpBuiltinExec`](https://tc39.github.io/ecma262/#sec-regexpbuiltinexec), we're supposed to `Get` `global` before `sticky`, so it won't take affect until `global` is used again.